File: Day3(give the sub array that sums upto a given number).cpp
#include <iostream>
using namespace std;

int main(){
	int n,x;
	cout<<"Enter the array size: ";
	cin>>n;
	int arr[n];
	cout<<"Enter the array: ";
	for(int i=0;i<n;i++){
		cin>>arr[i];
	}
	
	cout<<"Enter the sum: ";
	cin>>x;
	
	for(int i=0;i<n;i++){
		int sum=arr[i];
		if(sum==x){
			cout<<arr[i];
		}
		else{
			for(int j=i+1;j<n;j++){
				sum+=arr[j];
				if(sum==x){
					cout<<"Subarray is: ";
					for(int k=i; k<=j; k++){
						cout<<arr[k]<<" ";
					}
					cout<<"\n";
					return 0;
				}
			}	
		}
	}
	cout<<"No subarray found\n";
	return 0;
}

File: Day3(give the sub array that sums upto a given number).py
def findSubarray():
    n=int(input("Enter array size: "))
    arr=[]
    for i in range (n):
        l=int(input())
        arr.append(l)
    x=int(input("Enter sum: "))
    for i in range(0,n):
        sum=arr[i]
        if sum==x:
            arr[i]
        else:
            for j in range(i+1, n):
                sum+=arr[j]
                if sum==x:
                    print("The sub array is: ",arr[i:j+1])
                    return    
    print("No subarray found")
    
findSubarray()
